monitoring/mongodb/dashboard.py

```diff
@@ -0,0 +1,1397 @@
+        core.StatValueMappingItem(text="UNKNOWN", mapValue="6", color="", index=8),
+        core.StatValueMappingItem(text="ARBITER", mapValue="7", color="#8B8000", index=1),
+        core.StatValueMappingItem(text="DOWN", mapValue="8", color="", index=7),
+        core.StatValueMappingItem(text="ROLLBACK", mapValue="9", color="", index=6),
+        core.StatValueMappingItem(text="REMOVED", mapValue="10", color="", index=5),
+        core.StatValueMappingItem(text="Exporter is not connected", mapValue="null", color="#FF7383", index=2),
+    )
+]
+
+
+class MongoDBMetrics:
+    """MongoDB metrics expressions"""
+
+    @staticmethod
+    def up(job_pattern):
+        return f'sum(mongodb_up{{job="{job_pattern}"}})'
+
+    @staticmethod
+    def version_info():
+        return "avg by (service_name,mongodb) (mongodb_version_info{})"
+
+    @staticmethod
+    def db_stats(metric, filters=""):
+        base_filter = 'rs_nm=~"$rs_nm", rs_state="1"'
+        if filters:
+            base_filter += f", {filters}"
+        return f"max(mongodb_dbstats_{metric}{{{base_filter}}})"
+
+    @staticmethod
+    def replset_state(job_pattern):
+        return f'max by (pod) (mongodb_mongod_replset_my_state{{job=~"{job_pattern}"}})'
+
+    @staticmethod
+    def oplog_window():
+        return f"""min(
+  mongodb_mongod_replset_oplog_head_timestamp{{job=~"{JOBS["shard_pattern"]}"}}
+  -
+  mongodb_mongod_replset_oplog_tail_timestamp{{job=~"{JOBS["shard_pattern"]}"}}
+)"""
+
+    @staticmethod
+    def disk_usage():
+        return "avg(mongodb_dbstats_fsUsedSize{})/ avg (mongodb_dbstats_fsTotalSize{})"
+
+    @staticmethod
+    def cursors_open():
+        return 'sum by (job) (mongodb_mongod_metrics_cursor_open{rs_nm=~"$rs_nm"})'
+
+    @staticmethod
+    def connections(conn_type, pod_pattern):
+        return f'sum by(pod) (mongodb_ss_connections{{namespace="{NAMESPACE}", conn_type="{conn_type}", pod=~"{pod_pattern}"}})'
+
+    @staticmethod
+    def opcounters(op_types, rate_interval="[$__rate_interval]"):
+        return f'sum by(job) (rate(mongodb_ss_opcounters{{legacy_op_type=~"{op_types}", rs_nm=~"$rs_nm", job="{JOBS["shards"]}"}}{rate_interval}))'
+
+    @staticmethod
+    def replication_lag():
+        return f'last_over_time(mongodb_mongod_replset_member_replication_lag{{job=~"{JOBS["shard_pattern"]}"}}[$interval]) > 0'
+
+
+def mongodb_stat(title, expr="", unit="none", thresholds=None, targets=None, **kwargs):
+    """Create a MongoDB Stat panel with common settings"""
+    if targets is None and expr:
+        targets = [Target(expr=expr, legendFormat="__auto")]
+    elif targets is None:
+        targets = []
+
+    # Set default values if not provided in kwargs
+    defaults = {
+        "colorMode": "value",
+        "textMode": "auto",
+    }
+
+    # Update defaults with any provided kwargs
+    for key, default_value in defaults.items():
+        if key not in kwargs:
+            kwargs[key] = default_value
```

Comment:
not needed, can either use `|` to merge maps:
```
kwargs = defaults | kwargs
```

or simply expand both variables in the `Stat` call:
```
     Stat(
         title=title,
         dataSource=DATASOURCE,
         format=unit,
         thresholds=thresholds or THRESHOLDS["green_red"],
         targets=targets,
         **defaults,
         **kwargs,
     )
```